### PR TITLE
Add Vale linter

### PR DESCRIPTION
**Pull request by gewenyu99**
Hi friends, working to add a new linter in [Vale](https://vale.sh/explorer/).

## What I'm trying to do
Vale is a pretty extensible style checker for text files like Markdown, doc source, blogs, etc.
Lets you lint beyond spell checks, also for style, diction, etc.

I'm trying to add Vale as a linter so I can use it in `trunk check`.

## What I added
- [x] Vale as a tool, downloaded from GitHub releases
- [x] Vale as a linter, output as line and parsed with regex
- [x] A basic smoke tests for tool
- [x] Success case and an issue case tests for linter

## Test plan
- Tests described above
- Some manual testing by running Vale check with vale enabled.

### CODE REVIEW

File: linters/vale/vale.test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import * as path from "path";
+import { customLinterCheckTest, fuzzyLinterCheckTest } from "tests";
+import { TEST_DATA } from "tests/utils";
+import { createFuzzyMatcher } from "tests/utils/landing_state";
+import * as fs from "fs";
+import { FileIssue, LandingState } from "tests/types";
+
+// Specify a path and file to avoid accidentally
```
Comment (gewenyu99) [inline linters/vale/vale.test.ts:8]:
```suggestion
// Specify a path and file to avoid linting bad file
```

**Review by TylerJang27 — COMMENTED** [review]
Hi @gewenyu99! 👋 Thanks for the PR! Vale's been on our todo list for a while! I've added a couple stylistic notes and some cleanup. Let me know if you have any questions or other thoughts!

File: linters/vale/vale.test.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+const expectedBasicFileIssues = JSON.parse(
+  fs.readFileSync(path.resolve(__dirname, "expected_issues.json")).toString(),
+);
+
+fuzzyLinterCheckTest({
+  linterName: "vale",
+  testName: "poor spelling",
+  args: path.join(TEST_DATA, "documents/BAD_FILE.md"),
+  fileIssueAssertionCallback: createFuzzyMatcher(() => expectedBasicFileIssues as FileIssue[], 2),
+});
```
[the remark this answers is not in the document]
Comment (gewenyu99) [reply linters/vale/vale.test.ts:24]: Done

File: tools/vale/plugin.yaml
[the remark this answers is not in the document]
Comment (gewenyu99) [reply tools/vale/plugin.yaml:1]: Yep, done

File: tools/vale/plugin.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  - name: vale
+    downloads:
+      - os:
+          linux: Linux
+          macos: macOS
+          windows: Windows
+        cpu:
+          x86_64: 64-bit
+          arm_64: arm64
+        url: https://github.com/errata-ai/vale/releases/download/v${version}/vale_${version}_${os}_${cpu}.tar.gz
```
[the remark this answers is not in the document]
Comment (gewenyu99) [reply tools/vale/plugin.yaml:13]: Nice catch! Fixed

**Comment by gewenyu99** [comment]
Thanks for the review, wrapped some of this stuff up.

Would y'all appreciate a (separate) PR to the `CONTRIBUTING.md` and related files about some of the conventions we covered here? Save the next person some back and forth :)

**Review by TylerJang27 — APPROVED** [review]
Thanks for the PR! Excited to get this in. I'll add some copy to clear up `CONTRIBUTING.md`, it's been a while since it's been updated.

Did you go through the [linters quickstart](https://github.com/trunk-io/plugins/blob/main/CONTRIBUTING.md#linters) by chance? There's a Trunk Action that should autogenerate most of these files for the simple cases. Open to any other feedback that you might have!

**Comment by gewenyu99** [comment]
> Thanks for the PR! Excited to get this in. I'll add some copy to clear up `CONTRIBUTING.md`, it's been a while since it's been updated.
> 
> Did you go through the [linters quickstart](https://github.com/trunk-io/plugins/blob/main/CONTRIBUTING.md#linters) by chance? There's a Trunk Action that should autogenerate most of these files for the simple cases. Open to any other feedback that you might have!

Yeah I went through it, I saw the sentence about automatically created files but I couldn't figure out how to get there. After about 15 min I went MacGyver 😁 Wasn't difficult to figure out though

**Comment by TylerJang27** [comment]
Gotcha. I'll throw up a PR today or tomorrow to simplify things in that doc--it could use a facelift. I'll link it here when it's ready; feel free to add any thoughts!

**Comment by gewenyu99** [comment]
Perfect, was a lot of fun :D

**Comment by TylerJang27** [comment]
Put up #762 to revamp some of the quickstart setup. Feel free to chime in if you have any thoughts, especially with the linter setup: https://github.com/trunk-io/plugins/blob/tyler/contributing-revamp/CONTRIBUTING.md#linters (now includes links to the template files and some more intentional docs links)

**Pull request merged**